Update data-connect-troubleshooting.md (#9507)
microsoftgraph/microsoft-graph-docs-contrib: concepts/data-connect-troubleshooting.md

![Screenshot that shows an error for the already existent billing resource.](images/app-registration-already-premium-usage.png)

> ### Developer Email Missing

Reviewer: ```suggestion
### Developer email missing
```

> When creating a new registration, it may appear successful, but you’ll receive a "Developer Email Not Found" error. This happens when the registration is created with a guest user; using a non-guest user will succeed.

Reviewer: ```suggestion
When creating a new registration, it may appear successful, but you receive a "Developer email not found" error. This happens when the registration is created with a guest user; use a non-guest user to successfully complete the registration.
```
